src/contracts/core/DelegationManager.sol

```diff
@@ -701,54 +708,83 @@ contract DelegationManager is
+            require(startTimestamp + MIN_WITHDRAWAL_DELAY <= block.timestamp, WithdrawalDelayNotElapsed());
+            // source magnitudes from the time of completability
+            completableTimestamp = startTimestamp + MIN_WITHDRAWAL_DELAY;
+        }
+    }
+
+    /**
+     *
+     *                              SHARES CONVERSION FUNCTIONS
+     *
+     */
+
     /**
-     * @notice Withdraws `shares` in `strategy` to `withdrawer`. If the shares are virtual beaconChainETH shares, then a call is ultimately forwarded to the
-     * `staker`s EigenPod; otherwise a call is ultimately forwarded to the `strategy` with info on the `token`.
+     * @notice helper to calculate the new depositScalingFactor after adding shares. This is only used
+     * when a staker is depositing through the StrategyManager or EigenPodManager. A staker's depositScalingFactor
+     * is only updated when they have new deposits and their shares are being increased.
      */
-    function _withdrawSharesAsTokens(
+    function _updateDepositScalingFactor(
         address staker,
-        address withdrawer,
         IStrategy strategy,
-        uint256 shares,
-        IERC20 token
+        uint64 totalMagnitude,
+        Shares existingShares,
+        OwnedShares addedOwnedShares
     ) internal {
-        if (strategy == beaconChainETHStrategy) {
-            eigenPodManager.withdrawSharesAsTokens({podOwner: staker, destination: withdrawer, shares: shares});
+        uint256 newDepositScalingFactor;
+        if (existingShares.unwrap() == 0) {
+            newDepositScalingFactor = WAD / totalMagnitude;
         } else {
-            strategyManager.withdrawSharesAsTokens(withdrawer, strategy, shares, token);
+            // otherwise since
+            //
+            // newShares
+            //      = existingShares + addedShares
+            //      = existingPrincipalShares.toDelegatedShares(stakerScalingFactors[staker][strategy).toOwnedShares(totalMagnitude) + addedShares
+            //
+            // and it also is
+            //
+            // newShares
+            //     = newPrincipalShares.toDelegatedShares(stakerScalingFactors[staker][strategy).toOwnedShares(totalMagnitude)
+            //     = newPrincipalShares * newDepositScalingFactor / WAD * totalMagnitude / WAD 
+            //     = (existingPrincipalShares + addedShares) * newDepositScalingFactor / WAD * totalMagnitude / WAD 
+            //
+            // we can solve for
+            //
+            OwnedShares existingOwnedShares =
+                existingShares
+                .toDelegatedShares(depositScalingFactors[staker][strategy])
+                .toOwnedShares(totalMagnitude);
+            newDepositScalingFactor =
+                existingOwnedShares
+                .add(addedOwnedShares)
+                .unwrap()
+                .divWad(existingShares.unwrap() + addedOwnedShares.unwrap())
+                .divWad(totalMagnitude);
```

Comment:
I reviewed the Notion doc which lays out this calculation and I _think_ this is correct, but it really feels like this operation should probably live in a math library.
I believe we discussed this some in-person as well, but commenting here for tracking this.

Reply:
I believe this is fixed in #784 @ypatil12 ?